dace/transformation/passes/symbol_propagation.py

```diff
@@ -0,0 +1,214 @@
+                    str(s) in sdfg.arrays and isinstance(sdfg.arrays[str(s)], dt.View)
+                    for s in pystr_to_symbolic(v).free_symbols
+                ])
+            }
+
+            # Combine the symbols
+            if i == 0:
+                new_in_syms = sym_table
+            else:
+                self._combine_syms(new_in_syms, sym_table)
+
+        # Nested starting CFBGs should inherit the symbols from their parent
+        # Ignore SDFGs as nested SDFGs have symbol mappings
+        if (parent.start_block == cfgb and not isinstance(parent, SDFG)) or (isinstance(parent, ConditionalBlock)
+                                                                             and cfgb in parent.sub_regions()):
+            assert new_in_syms == {}
+            new_in_syms = in_syms[parent]
+
+            # For LoopRegions, remove loop carried variables from the incoming symbols
+            if isinstance(parent, LoopRegion):
+                new_in_syms = copy.deepcopy(new_in_syms)
+                all_syms = set([s for e in parent.all_interstate_edges() for s in e.data.assignments.keys()])
+                for sym in all_syms:
+                    if sym in new_in_syms:
+                        new_in_syms[sym] = None
+
+        return new_in_syms
+
+    # Given a CFGB, builds the outgoing set of symbols
+    def _get_out_syms(
+        self,
+        cfgb: ControlFlowBlock,
+        parent: ControlFlowRegion,
+        in_syms: Dict[ControlFlowBlock, Dict[str, Any]],
+        out_syms: Dict[ControlFlowBlock, Dict[str, Any]],
+    ) -> Dict[str, Any]:
+        if isinstance(cfgb, LoopRegion):
+            # Any symbol that is assigned in the loop region is not propagated out
+            new_out_syms = copy.deepcopy(in_syms[cfgb])
+            for edge in cfgb.all_interstate_edges():
+                for sym in edge.data.assignments.keys():
+                    if sym in new_out_syms:
+                        new_out_syms[sym] = None
+            return new_out_syms
+
+        elif isinstance(cfgb, ConditionalBlock):
+            # Combine all outgoing symbols of the branches
+            new_out_syms = copy.deepcopy(out_syms[cfgb.sub_regions()[0]])
+            for b in cfgb.sub_regions():
+                self._combine_syms(new_out_syms, out_syms[b])
+
+            # If no else branch is present, also combine the incoming table (implicit else branch)
+            has_non_conds = any([c is None for c, _ in cfgb.branches])
+            if not has_non_conds:
+                self._combine_syms(new_out_syms, in_syms[cfgb])
+
+            return new_out_syms
+
+        elif isinstance(cfgb, SDFGState):
+            # Cannot change symbols in SDFGStates
+            return in_syms[cfgb]
+
+        else:
+            # Use sink symbols as outgoing symbols
+            sink_nodes = [n for n in cfgb.nodes() if cfgb.out_degree(n) == 0 and isinstance(n, ControlFlowBlock)]
```

Comment:
2 things:
1. This seems to not consider non-sink exits, such as return, continue, break, correct?
2. Why is this handled separately from loop regions?